pkg/prometheus/agent/operator.go

```diff
@@ -1344,6 +1344,11 @@ func (c *Operator) checkAgentAuthorization(ctx context.Context) (bool, error) {
 	var ssarResponse *authv1.SelfSubjectAccessReview
 	var err error
 
+	missingPermissions := map[string][]string{
+		monitoringv1alpha1.PrometheusAgentName:                           {},
+		fmt.Sprintf("%s/status", monitoringv1alpha1.PrometheusAgentName): {},
+	}
```

Comment:
wouldn't it better if we kept it empty and add `missing[resource] = append(missing[resource], verb)` since append handles nil fine. 

```
	missing := map[string][]string{}
	missing["a"] = append(missing["a"], "bar")
```
works fine

Reply:
Oh awesome, thanks for the suggestion :)

Indeed it looks a lot cleaner this way